.github/workflows/bump-version.yml

```diff
@@ -9,12 +9,13 @@ jobs:
       runs-on: ubuntu-latest
   
       env:
-        GH_PUSH_TOKEN: ${{ secrets.GH_PUSH_TOKEN || github.token }}
+        GH_PUSH_TOKEN: ${{ secrets.GH_PUSH_TOKEN }}
```

Comment:
The idea of coallescing to `github.token` was to prevent the checkout step from erroring when the token is missing, but then the push step errors and we don't have a way to tell whether the token was provided by the secret or the default. So instead, I'm removing the default value for the `env.GH_PUSH_TOKEN` and adding a conditional for the checkout step